client/src/controllers/TagController.test.js

```diff
@@ -64,13 +79,218 @@ describe('TagController', () => {
     expect(preprocessTag("'long black'")).toEqual(`"'long black'"`);
     expect(preprocessTag('caffe latte')).toEqual(`"caffe latte"`);
 
-    // check the custom clear behaviour
+    // check the custom clear behavior
     document
       .getElementById('id_tags')
       .dispatchEvent(new CustomEvent('example:event'));
 
-    await new Promise(requestAnimationFrame);
+    await flushPromises();
 
     expect(tagitMock).toHaveBeenCalledWith('removeAll');
   });
+
+  describe('autocomplete', () => {
+    let autocompleteSource;
+
+    beforeAll(async () => {
+      document.body.innerHTML = `
+      <form id="form">
+        <input
+          id="id_tags"
+          type="text"
+          name="tags"
+          data-controller="w-tag"
+          data-action="example:event->w-tag#clear"
+          data-w-tag-options-value='${JSON.stringify({
+            allowSpaces: true,
+            tagLimit: 10,
+          })}'
+          data-w-tag-url-value="/admin/tag-autocomplete/"
+        >
+      </form>`;
+
+      await flushPromises();
+
+      autocompleteSource = tagitMock.mock.calls[0][0].autocomplete.source;
```

Comment:
This had a small issue in it, we should be getting the last call, as it's run in previous tests.

```suggestion
      // get last mock call
      autocompleteSource =
        tagitMock.mock.calls[tagitMock.mock.calls.length - 1][0].autocomplete
          .source;
          
```